File: app/src/main/java/com/cartenz/cartenztaxcore/feature/base/BaseFragment.java
package com.cartenz.cartenztaxcore.feature.base;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseFragment extends com.cartenz.core.BaseFragment {
    private Unbinder unbinder;

    @Override
    public void initBaseCreateView() {
        unbinder = ButterKnife.bind(this, view);
        initCreateView();
    }

    protected abstract void initCreateView();

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }


}

File: kotlinapp/src/main/java/com/cartenz/kotlinapp/feature/base/BaseFragment.kt
package com.cartenz.kotlinapp.feature.base


abstract class BaseFragment : com.cartenz.kotlin_core.CartenzBaseFragment() {

    override fun initBaseCreateView() {
        initCreateView()
    }

    protected abstract fun initCreateView()

    override fun onDestroyView() {
        super.onDestroyView()
    }

}
